modules/mauth-common/src/main/java/com/mdsol/mauth/util/MAuthSignatureHelper.java

```diff
@@ -35,18 +34,18 @@
    * @param httpMethod: Http_Verb
    * @param resourceUrl: resource_url_path (no host, port or query string; first "/" is included)
    * @param queryParameters: request parameters string
-   * @param requestBody: request body string
+   * @param requestBody: request body in byte[]
    * @param epochTime: current seconds since Epoch
    * @param mauthVersion: Mauth version (MSW or MSWV2)
    * @return String
    *
    * @throws MAuthSigningException
    */
   public static String generateStringToSign(UUID appUUID, String httpMethod, String resourceUrl, String queryParameters,
-      String requestBody, String epochTime, MAuthVersion mauthVersion) throws MAuthSigningException {
```

Comment:
This breaks binary compatibility. Should leave this method alone (with deprecation)

Reply:
**generateStringToSign** is a new method to support the both V1 and V2.  **generateUnencryptedSignature** is the original one for V1 and @Deprecated already

Reply:
Regarding requestBody is  an array of bytes,  the implementation 
 of **generateUnencryptedSignature** doesn't support non-UTF-8 string and I just leave as it is. But we do support binary body for V2 in **generateStringToSignV2**

Reply:
The public signature of `generateStringToSign` is still broken though..I can pass a non UTF-8 byte array with MWS V1 and it'll blow up. Looking at the usages, all call sites already know what the MWS version is, so do we need this helper function at all?

Reply:
Yeah, if it is V1, generateStringToSign calls generateUnencryptedSignature that is existing method for V1 and doesn't  support non UTF-8 byte array.
It seems some Java modules are using this function.

Reply:
Yes that's my point - we're introducing a potential issue here with the API as it is now.

Let's look at all usages of `generateStringToSign`

* com.mdsol.mauth.RequestAuthenticator#validateSignatureV2: hardcoded to V2
* com.mdsol.mauth.scaladsl.RequestAuthenticator#validateSignatureV2: hardcoded to V2
* com.mdsol.mauth.DefaultSigner#generateRequestHeadersV2: hardcoded to V2

So there's no need for generateStringToSign at all. Any of those use sites could've call `generateStringToSignV2` directly.

Reply:
removed generateStringToSign  as suggested. See https://github.com/mdsol/mauth-jvm-clients/pull/92/commits/1105451ea7c6fdbab0893eae57da45edac6d2021